### PR TITLE
glance: run db sync and load metadefs

### PR DESCRIPTION
**Pull request by cmurphy**
In bdf0cce we removed the glance-registry recipe which also removed the
glance-manage db_sync and glance-manage db_load_metadefs executions. We
didn't notice this was a problem at first because it seems that the
glance API service is capable of running the db sync itself, but the
metadef loading was still not happening. This causes tempest failures
because the contents of GET /v2/metadefs/resource_types will be empty.
This patch fixes the issue by adding back this setup steps for the API
service.

### CODE REVIEW

File: chef/cookbooks/glance/recipes/api.rb [hunk 1]
```diff
@@ -67,7 +67,7 @@
   source "glance-api.conf.erb"
   owner "root"
   group node[:glance][:group]
-  mode 0640
+  mode 0o640
```
Comment (toabctl) [inline chef/cookbooks/glance/recipes/api.rb]: why this change?

Comment (cmurphy) [reply chef/cookbooks/glance/recipes/api.rb]: My mistake, hound was angry about the other template and I fixed the wrong one.

Comment (cmurphy) [reply chef/cookbooks/glance/recipes/api.rb]: This is fixed, can you re-review?
